lib/Cro/WebApp/Form.pm6

```diff
@@ -385,6 +390,8 @@ role Cro::WebApp::Form {
         }
         for self.^attributes.grep(*.has_accessor) -> Attribute $attr {
             my ($control-type, %properties) = self!calculate-control-type($attr);
+            die "Can't have type 'file' in a GET form"
+                if $control-type eq 'file' && $method eq 'get';
```

Comment:
Slight preference for a typed exception, and also include the class name of the form perhaps.

Reply:
done